stdparams.go

```diff
@@ -0,0 +1,38 @@
+// +build go1.7
+
+package httprouter
+
+import (
+	"context"
+	"net/http"
+)
+
+type paramsKey struct{}
```

Comment:
Because this is extreme micro optimization that will only save you 4 bytes during the whole runtime of the program because the only value of this type is ever only allocated once during package initialisation. I do not think this is worth the disadvantages I mentioned earlier: maybe more confusing for new devs, deviates from the standard, it triggers more discussion than the whole topic is worth :smile: